src/tests/mam4_ndrop_unit_tests.cpp

```diff
@@ -21,132 +21,184 @@
   }
+}
 
-  // Call mam4xx kernel across all modes, levels
-  Kokkos::parallel_for(
-      "compute_dry_particle_size", nlev, KOKKOS_LAMBDA(const int i) {
-        mode_avg_dry_particle_diam(diags, progs, i);
-      });
-
-  mam4::Real naerosol[nmodes];
+TEST_CASE("test_qsat", "mam4_ndrop_unit_tests") {
+  ekat::Comm comm;
+  ekat::logger::Logger<> logger("ndrop unit tests",
+                                ekat::logger::LogLevel::debug, comm);
 
-  // iterate over modes and columns to calculate aer num
-  for (int m = 0; m < nmodes; m++) {
-    auto h_diam_total =
-        Kokkos::create_mirror_view(diags.dry_geometric_mean_diameter_total[m]);
-    Kokkos::deep_copy(h_diam_total, diags.dry_geometric_mean_diameter_total[m]);
-
-    for (int k = 0; k < nlev; k++) {
-      logger.info("k = {}, m = {}", k, m);
-      Real vaerosol = conversions::mean_particle_volume_from_diameter(
-          h_diam_total(k), modes(m).mean_std_dev);
-      logger.info("vaerosol = {}", vaerosol);
-
-      Real num2vol_ratio_min =
-          1.0 / conversions::mean_particle_volume_from_diameter(
-                    modes(m).min_diameter, modes(m).mean_std_dev);
-      Real num2vol_ratio_max =
-          1.0 / conversions::mean_particle_volume_from_diameter(
-                    modes(m).max_diameter, modes(m).mean_std_dev);
-      Real min_bound = vaerosol * num2vol_ratio_max;
-      Real max_bound = vaerosol * num2vol_ratio_min;
-
-      logger.info("atm.temperature = {}, atm.pressure = {}", h_T(k), h_P(k));
-      Real middle = (number_mixing_ratio + number_mixing_ratio) *
-                    conversions::density_of_ideal_gas(h_T(k), h_P(k));
-
-      ndrop::get_aer_num(diags, progs, atm, m, k, naerosol);
-      logger.info("naerosol[{}] = {}", m, naerosol[m]);
-
-      logger.info("min bound = {}, max bound = {}", min_bound, max_bound);
-      logger.info("progs calc = {}", middle);
-
-      // TODO: this test needs to be revisited to run properly on GPU
-      //  come back when this function is being used in a ported
-      //  parameterization
-
-      REQUIRE(
-          FloatingPoint<Real>::in_bounds(naerosol[m], min_bound, max_bound));
-      bool check_calc = FloatingPoint<Real>::equiv(naerosol[m], min_bound) ||
-                        FloatingPoint<Real>::equiv(naerosol[m], middle) ||
-                        FloatingPoint<Real>::equiv(naerosol[m], max_bound);
-      REQUIRE(check_calc);
-    }
+  std::ostringstream ss;
+
+  // I took these from the mam_x_validation file dropmixnuc_ts_1417.yaml
+  // temperature [K]
+  const Real t[AeroConfig::num_modes()] = {272.15, 0.2529892939E003,
+                                           0.2159618556E003, 0.2686359632E003};
+  // pressure [Pa]
+  const Real p[AeroConfig::num_modes()] = {101325.0, 0.1238254131E002,
+                                           0.1999170945E005, 0.9524671355E005};
+
+  for (int i = 0; i < AeroConfig::num_modes(); ++i) {
+    ss << "i = " << i;
```

Comment:
Not a big deal, but you might find the [fmt](https://github.com/fmtlib/fmt) library a little more convenient for creating log messages. EKAT provides `fmt` in addition to our logging library, so feel free to use it.